File: kotlin/src/test/kotlin/io/icednut/leetcode/exercise/day00/binarytree/SolutionTest.kt
package io.icednut.leetcode.exercise.day00.binarytree

import org.junit.jupiter.api.Assertions.assertArrayEquals
import org.junit.jupiter.api.Test

/**
 * @author will.109
 * @date 2020/04/16
 */
internal class SolutionTest {

    @Test
    fun `binary tree에 대해 Inorder 순회를 할 수 있어야 한다`() {
        val solution = Solution()
        val root = TreeNode(1)
        val right = TreeNode(2)
        right.left = TreeNode(3)
        root.right = right

        val result = solution.inorderTraversal(root)

        assertArrayEquals(result.toIntArray(), intArrayOf(1, 3, 2))
    }
}

File: scala/src/test/scala/io/icednut/algorithm_exercise/leetcode/q153/SolutionTest.scala
package io.icednut.algorithm_exercise.leetcode.q153

import org.scalatest.funsuite.AnyFunSuite

class SolutionTest extends AnyFunSuite {

  test("case 01") {
    val result = Solution.findMin(Array(3, 4, 5, 1, 2))

    assert(result == 1)
  }

  test("case 02") {
    val result = Solution.findMin(Array(4, 5, 6, 7, 0, 1, 2))

    assert(result == 0)
  }

  test("case 03") {
    val result = Solution.findMin(Array(11, 13, 15, 17))

    assert(result == 11)
  }
}
